Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/BaseDAO.java

```diff
@@ -235,7 +235,11 @@ public List<T> getByQuery(String query, Map<String, Object> parameters) {
      * @param query
      *            as String
      * @return list of beans objects
+     * @deprecated Form the query with placeholders and pass parameters. Use
+     *             {@code BeanQuery(Bean.class)} and
+     *             {@link #getByQuery(String, Map)}.
      */
+    @Deprecated
```

Comment:
Why is this method marked as `deprecated` but still in (heavy) use inside the application? Should not first all existing calls be replaced and later this method marked as deprecated or even better removed if not used? Marking a method as deprecated creates CodeQL warnings which are correct and right.

Reply:
I set the method to `@Deprecated` because it doesn't allow parameter passing. The parameters are hard-coded into the query. The existence of the method can tempt programmers to implement queries using string concatenation instead of using placeholders and passing the values ​​separately. Therefore, I think it would be a good idea to eliminate this method altogether in the long term. In the places where it is currently used, it is safe because the entire string is hard-coded and no user input is inserted, but I still think it should be eliminated in the future. It's also explained here in the Javadoc. That's why I set the method to `@Deprecated`. Rewriting the existing uses should be the subject of a separate pull request.